service/controller/appcatalog/resource/appcatalogentry/create.go

```diff
@@ -168,154 +170,178 @@ func (r *Resource) updateAppCatalogEntry(ctx context.Context, entryCR *v1alpha1.
-					if err != nil {
-						return nil, microerror.Mask(err)
-					}
-				}
-			}
+	if m != nil {
+		entryCR.Annotations = m.Annotations
+		entryCR.Spec.Chart.APIVersion = m.ChartAPIVersion
+		entryCR.Spec.Restrictions = m.Restrictions
+		entryCR.Spec.DateCreated = m.DataCreated
+		entryCR.Spec.DateUpdated = m.DataCreated
+	}
 
-			var isLatest bool
+	if entryCR.Spec.Chart.APIVersion == "" {
+		// chartAPIVersion default is `v1`.
+		entryCR.Spec.Chart.APIVersion = "v1"
+	}
 
-			// We set the latest label to true for easier filtering.
-			if e.Version == latestVersion {
-				isLatest = true
-			}
+	if entryCR.Spec.DateCreated == nil {
+		// If meta.yaml does not have dateCreated, use the timestamp from app.
+		entryCR.Spec.DateCreated = &e.Created
+		entryCR.Spec.DateUpdated = updatedTime
+	}
 
-			entryCR := &v1alpha1.AppCatalogEntry{
-				TypeMeta: metav1.TypeMeta{
-					APIVersion: apiVersion,
-					Kind:       kindAppCatalogEntry,
-				},
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      name,
-					Namespace: metav1.NamespaceDefault,
-					Labels: map[string]string{
-						label.AppKubernetesName:    e.Name,
-						label.AppKubernetesVersion: e.Version.String(),
-						label.CatalogName:          cr.Name,
-						label.CatalogType:          key.AppCatalogType(cr),
-						pkglabel.Latest:            strconv.FormatBool(isLatest),
-						label.ManagedBy:            key.AppCatalogEntryManagedBy(project.Name()),
-					},
-					OwnerReferences: []metav1.OwnerReference{
-						{
-							APIVersion:         apiVersion,
-							BlockOwnerDeletion: to.BoolP(true),
-							Kind:               kindAppCatalog,
-							Name:               cr.Name,
-							UID:                cr.UID,
-						},
-					},
-				},
-				Spec: v1alpha1.AppCatalogEntrySpec{
-					AppName:    e.Name,
-					AppVersion: e.AppVersion,
-					Catalog: v1alpha1.AppCatalogEntrySpecCatalog{
-						Name: cr.Name,
-						// Namespace will be empty until appcatalog CRs become namespace scoped.
-						Namespace: "",
-					},
-					Chart: v1alpha1.AppCatalogEntrySpecChart{
-						Home: e.Home,
-						Icon: e.Icon,
-					},
-					Version: e.Version.String(),
-				},
-			}
+	return entryCR, nil
+}
 
-			if m != nil {
-				entryCR.Annotations = m.Annotations
-				entryCR.Spec.Chart.APIVersion = m.ChartAPIVersion
-				entryCR.Spec.Restrictions = m.Restrictions
-				entryCR.Spec.DateCreated = m.DataCreated
-				entryCR.Spec.DateUpdated = m.DataCreated
-			}
+// getLatestEntry returns the entry with the highest version without considering the creation date.
+func (r *Resource) getLatestEntry(ctx context.Context, entries []entry) (entry, error) {
+	var latestIndex int
+	var latestVersion semver.Version
+	var latestCreated *metav1.Time
+
+	for i := 0; i < len(entries); i++ {
+		v, err := semver.NewVersion(entries[i].Version)
+		if errors.As(err, &semver.ErrInvalidSemVer) {
+			r.logger.Debugf(ctx, "invalid semver from converting app entries %s, version is %s", entries[i].Name, entries[i].Version)
```

Comment:
```suggestion
			r.logger.Debugf(ctx, "invalid semver from converting app entry %s, version is %s", entries[i].Name, entries[i].Version)
```

I think singular instead of plural is better. As the log is just for this entry.